modules/programs/git.nix

```diff
@@ -447,6 +465,19 @@ in {
         gpg.program = cfg.signing.gpgPath;
       };
     })
+    (mkIf (cfg.hooks != { }) {
+      programs.git.iniContent = {
+        core.hooksPath = "${pkgs.symlinkJoin {
```

Comment:
I have tried to add hooks to my configuration in the past, and ran into a problem with symlinks. The thing is that when the hooks directory contains symlinks to the nix store, when git initializes a new repository it will just copy the symlinks to `.git/hooks` instead of the actual files. Then when I changed a hook and subsequently ran a nix garbage collect, the symlinks in various `.git/hooks` directories suddenly didn't point to a script anymore because they were gc'ed. So for this to work properly, the path that `core.hooksPath` points to should contain only regular files, not symlinks to the nix store.

Reply:
After writing the above I looked up my config and it instead uses the `init.templatedir` setting, not `core.hooksPath`. So I don't think the above actually applies, as it seems `init.templatedir` is copied to `.git/hooks` on `git init` while `core.hooksPath` isn't and so doesn't suffer from the dead symlink problem.

I'm not sure, though, if this should use `core.hooksPath` instead of `init.templatedir`, as the former prevents one to use hooks on a per-repository basis.

Reply:
PS this is the config I'm using, where `./hooks` is a directory containing all my hooks:

```nix
{ pkgs, ... }: {
  config.git.extraConfig.init.templatedir =
    "${pkgs.runCommandNoCCLocal "git-hooks" {} ''
      mkdir -p $out/hooks
      cp -r "${./hooks}"/* $out/hooks
    ''}";
```

Reply:
I agree that for certain setups, `templatedir` is the appropriate place to put hooks. That said, for a top-level `hooks` configuration, home-manager placing them in templateDir that would be non-obvious behavior in my view.

Reply:
>  core.hooksPath isn't and so doesn't suffer from the dead symlink problem

I can confirm that usage of `core.hooksPath` works as intended. The symlink problem is not an issue.